src/core/Akka/Util/ByteIterator.cs

```diff
@@ -496,7 +500,11 @@ public bool IsEmpty
 
         public T Head
         {
-            get { return _array[_offset]; }
+            get
+            {
+                if (!IsEmpty) return _array[_offset];
+                return default(T);
```

Comment:
I think head should throw if the list is empty, I would expect the same behavior as with First. Also suggests the name offset that it could throw indexoutofrange even if the list isn't empty, offset >= count